api/konnect/v1alpha1/konnect_gateway_controlplane_conversion.go

```diff
@@ -0,0 +1,95 @@
+package v1alpha1
+
+import (
+	"github.com/samber/lo"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+
+	konnectv1alpha2 "github.com/kong/kubernetes-configuration/v2/api/konnect/v1alpha2"
+)
+
+// ConvertTo converts this KonnectGatewayControlPlane (v1alpha1) to the Hub version (v1alpha2).
+func (src *KonnectGatewayControlPlane) ConvertTo(dstRaw conversion.Hub) error {
+
+	dst := dstRaw.(*konnectv1alpha2.KonnectGatewayControlPlane)
+	dst.ObjectMeta = src.ObjectMeta
+
+	// Convert the changed fields between v1alpha1 and v1alpha2.
+	dst.Spec.CreateControlPlaneRequest = createControlplaneRequestFromSpec(src.Spec)
+
+	if src.Spec.Mirror != nil {
+		dst.Spec.Mirror = &konnectv1alpha2.MirrorSpec{
+			Konnect: konnectv1alpha2.MirrorKonnect{
+				ID: src.Spec.Mirror.Konnect.ID,
+			},
+		}
+	} else {
+		dst.Spec.Mirror = nil
+	}
+	dst.Spec.Source = src.Spec.Source
```

Comment:
Should we use `DeepCopy` method or other dedicated ways here for pointer values and slices?

Reply:
None of this fields have a dedicated `DeepCopy` method.

As far as I can tell, it's safe to assume that the `source` object is destroyed at the end of the conversion, making the `destination` object the owner of all references assigned to it. We expect that the pointer/complex structs assignments are safe to do without the need to `deepCopy` them. 
https://github.com/kubernetes-sigs/controller-runtime/blob/c20ea143a236a34fb331e6c04820b75aac444e7d/pkg/webhook/conversion/conversion.go#L90-L149